openwisp_notifications/static/openwisp-notifications/js/preferences.js

```diff
@@ -0,0 +1,445 @@
+'use strict';
+
+if (typeof gettext === 'undefined') {
+    var gettext = function(word){ return word; };
+}
+
+(function ($) {
+    let isUpdateInProgress = false;
+
+    $(document).ready(function () {
+        const userId = $('.settings-container').data('user-id');
+        fetchGlobalSettings(userId);
+    });
+
+    function fetchGlobalSettings(userId) {
+        $.getJSON(`/api/v1/notifications/user/${userId}/preference/`, function (globalData) {
+            const isGlobalWebChecked = globalData.web;
+            const isGlobalEmailChecked = globalData.email;
+
+            $('#global-web').prop('checked', isGlobalWebChecked);
+            $('#global-email').prop('checked', isGlobalEmailChecked);
+
+            initializeGlobalSettingsEventListener(userId);
+
+            fetchNotificationSettings(userId, isGlobalWebChecked, isGlobalEmailChecked);
+        }).fail(function () {
+            showToast('error', gettext('Error fetching global settings. Please try again.'));
+        });
+    }
+
+    function fetchNotificationSettings(userId, isGlobalWebChecked, isGlobalEmailChecked) {
+        let allResults = [];
+        let currentPage = 1;
+
+        (function fetchPage() {
+            $.getJSON(`/api/v1/notifications/user/${userId}/user-setting/?page_size=100&page=${currentPage}`, function (data) {
```

Comment:
Can you format the URL as we do it for the notification widget?, i.e prepend the `notificationApiHost`.

https://github.com/openwisp/openwisp-notifications/blob/63a29862e97faa1cc5847a149189be06171ee564/openwisp_notifications/static/openwisp-notifications/js/notifications.js#L316

Reply:
Updated.